### PR TITLE
Taylor Gentry: completed string katas by implementing functions to pass tests

### PR DESCRIPTION
**Pull request by tgentry300**

### CODE REVIEW

File: string1.py [hunk 1]
```diff
@@ -25,7 +25,11 @@
 # and donuts(23) returns 'Number of donuts: many'
 def donuts(count):
     """Your code goes here.  Edit this docstring."""
```
Comment (madarp) [inline string1.py]: Edit the docstring .. please :)

File: string1.py [hunk 2]
```diff
@@ -25,7 +25,11 @@
...
-    return
+    if count < 10:
+        return(str('Number of donuts: ' + str(count)))
+    else:
+        return('Number of donuts: many')
+
```
Comment (madarp) [inline string1.py]: Recommend to try out string formatting syntax:
(you compute result ...)
return 'Number of donuts: {}'.format(result)

File: string1.py [hunk 3]
```diff
@@ -35,7 +39,10 @@ def donuts(count):
 # is less than 2, return instead the empty string.
 def both_ends(s):
     """Your code goes here.  Edit this docstring."""
-    return
+    if len(s)<2:
+        return ''
+    else:
+        return s[0:2] + s[-2:]
```
Comment (madarp) [inline string1.py]: YES

File: string2.py [hunk 4]
```diff
@@ -19,6 +19,12 @@
 
 def verbing(s):
     """Your code goes here.  Edit this docstring."""
+    if len(s) < 3:
+        return s
+    elif s[-3:] == 'ing':
+        return s + 'ly'
+    else:
+        return s + 'ing'
     return
```
Comment (madarp) [inline string2.py]: No bare returns.
Here you should just return s + 'ing', outside of the if-else clause

File: string2.py [hunk 5]
```diff
@@ -19,6 +19,12 @@
...
```
Comment (madarp) [inline string2.py]: Don't need the bare return here -- they are implied if you're not returning anything.

File: string2.py [hunk 6]
```diff
@@ -42,9 +53,18 @@ def not_bad(s):
 # e.g. 'abcde', the front half is 'abc', the back half 'de'.
 # Given 2 strings, a and b, return a string of the form
 #  a-front + b-front + a-back + b-back
+
+#This function splits a string in half and adds the 'remaining character' to the front half of the string if len(string) is odd.
+def split(s):
+    half, remaining = divmod(len(s), 2)
+    return s[:half + remaining], s[half + remaining:]
+
 def front_back(a, b):
```
Comment (madarp) [inline string2.py]: Be careful .. defining functions with same name as built-in python methods e.g. 'split()' will confuse readers.
However, Nice use of helper function!

File: string2.py (hunk 4 above)
Comment (tgentry300) [reply string2.py]: I think I just didn't delete the one that was already there.  I can do that.

File: string1.py (hunk 1 above)
Comment (tgentry300) [reply string1.py]: When editing the docstring, what should we put in there?

Comment (madarp) [reply string1.py]: Why, I am delighted you asked!! Just a one-liner about what the function does, if it's not obvious from the function name itself.  It's a habit to cultivate.  More about: https://stackoverflow.com/questions/601900/how-to-write-meaningful-docstrings

**Comment by madarp** [comment]
closing as complete

**Pull request closed without merging**